hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/ContractBurnHTSV2SecurityModelSuite.java

```diff
@@ -0,0 +1,693 @@
+                        getTxnRecord(SIGNER_AND_PAYER_ARE_DIFFERENT)
+                                .andAllChildRecords()
+                                .hasChildRecords(recordWith().status(SUCCESS)));
+    }
+
+    @HapiTest
+    final HapiSpec V2Security004FungibleTokenBurnNegative() {
+        final var initialAmount = 20L;
+        final var amountToBurn = 5L;
+        final AtomicReference<TokenID> fungible = new AtomicReference<>();
+
+        return propertyPreservingHapiSpec("V2Security004FungibleTokenBurnNegative")
+                .preserving(CONTRACTS_MAX_NUM_WITH_HAPI_SIGS_ACCESS)
+                .given(
+                        overriding(CONTRACTS_MAX_NUM_WITH_HAPI_SIGS_ACCESS, CONTRACTS_V2_SECURITY_MODEL_BLOCK_CUTOFF),
+                        cryptoCreate(TOKEN_TREASURY),
+                        cryptoCreate(SIGNER).balance(ONE_MILLION_HBARS),
+                        tokenCreate(FUNGIBLE_TOKEN)
+                                .tokenType(TokenType.FUNGIBLE_COMMON)
+                                .initialSupply(initialAmount)
+                                .treasury(TOKEN_TREASURY)
+                                .adminKey(TOKEN_TREASURY)
+                                .supplyKey(TOKEN_TREASURY)
+                                .exposingCreatedIdTo(idLit -> fungible.set(asToken(idLit))),
+                        uploadInitCode(HTS_CALLS),
+                        contractCreate(HTS_CALLS),
+                        uploadInitCode(MINT_CONTRACT),
+                        sourcing(() -> contractCreate(
+                                MINT_CONTRACT, HapiParserUtil.asHeadlongAddress(asAddress(fungible.get())))))
+                .when(withOpContext((spec, opLog) -> allRunFor(
+                        spec,
+                        // Test Case 1: Signer paying and signing a token burn transaction,
+                        // SIGNER → call → CONTRACT → call → PRECOMPILE
+                        // The signer and the token don't have updated keys
+                        contractCall(
+                                HTS_CALLS,
+                                "burnTokenCall",
+                                HapiParserUtil.asHeadlongAddress(
+                                        asAddress(spec.registry().getTokenID(FUNGIBLE_TOKEN))),
+                                BigInteger.valueOf(amountToBurn),
+                                new long[0])
+                                .via(SIGNER_AND_TOKEN_HAVE_NO_UPDATED_KEYS)
+                                .gas(GAS_TO_OFFER)
+                                .payingWith(SIGNER)
+                                .signedBy(SIGNER),
+                        // verify that the total supply of the tokens is not affected
+                        getTokenInfo(FUNGIBLE_TOKEN).hasTotalSupply(initialAmount),
+                        getTxnRecord(SIGNER_AND_TOKEN_HAVE_NO_UPDATED_KEYS)
+                                .andAllChildRecords()
+                                .logged(),
```

Comment:
It would be a good idea to use .hasKnownStatus(CONTRACT_REVERT_EXECUTED) check for each contractCall so that it is more readable what is the top level transaction status.

Reply:
fixed